### PR TITLE
Move Slack notifications into separated class

### PR DESCRIPTION
**Pull request by eagletmt**
I've refactored Slack notifications for the next change.

@cookpad/dev-infra @k0kubun 👓

### CODE REVIEW

**Comment by eisuke** [comment]
I think It is better to ignore the occurring error if the notification fails. (e.g. Slack API is down.)
However, there is a disadvantage that debugging becomes difficult...
What do you think?

File: lib/barbeque/slack_notifier.rb [hunk 1]
```diff
@@ -0,0 +1,71 @@
+require 'barbeque/slack_client'
+
+module Barbeque
+  class SlackNotifier
```
Comment (k0kubun) [inline lib/barbeque/slack_notifier.rb]: [nits] As it doesn't have ivar like `SlackClient` instance or some state for now, I would define this as module and define all methods as class methods to make it clear that it doesn't have such things.

Comment (eagletmt) [inline lib/barbeque/slack_notifier.rb]: I did so https://github.com/cookpad/barbeque/pull/31/commits/4625471b5fc3a800da753ed82ceb7a5b8b0616e1

**Comment by eagletmt** [comment]
> I think It is better to ignore the occurring error if the notification fails.

I agree but it's out of scope of this patch. This patch is just refactoring. No observable behaviors will change.

**Pull request merged**